sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py

```diff
@@ -270,6 +291,217 @@ def create_experiment(
+        },
+        ValueError,
+    ),
+    (
+        "set both options",
+        {
+            "name": "tune_test",
+            "objective": lambda x: x,
+            "model_provider_parameters": HuggingFaceModelParams(),
+        },
+        ValueError,
+    ),
+    (
+        "set no option",
+        {
+            "name": "tune_test",
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only objective",
+        {
+            "name": "tune_test",
+            "objective": lambda x: x,
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only parameters",
+        {
+            "name": "tune_test",
+            "parameters": {"param": "value"},
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only model_provider_parameters",
+        {
+            "name": "tune_test",
+            "model_provider_parameters": HuggingFaceModelParams(),
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only dataset_provider_parameters",
+        {
+            "name": "tune_test",
+            "dataset_provider_parameters": HuggingFaceDatasetParams(),
+        },
+        ValueError,
+    ),
+    (
+        "set one option but missing parameters - only trainer_parameters",
+        {
+            "name": "tune_test",
+            "trainer_parameters": HuggingFaceTrainerParams(),
+        },
+        ValueError,
+    ),
+    (
+        "invalid env_per_trial",
+        {
+            "name": "tune_test",
+            "objective": lambda x: x,
+            "parameters": {"param": "value"},
+            "env_per_trial": "invalid",
+        },
+        ValueError,
+    ),
+    (
+        "invalid model_provider_parameters",
+        {
+            "name": "tune_test",
+            "model_provider_parameters": "invalid",
+            "dataset_provider_parameters": HuggingFaceDatasetParams(),
+            "trainer_parameters": HuggingFaceTrainerParams(),
+        },
+        ValueError,
+    ),
+    (
+        "invalid dataset_provider_parameters",
+        {
+            "name": "tune_test",
+            "model_provider_parameters": HuggingFaceModelParams(),
+            "dataset_provider_parameters": "invalid",
+            "trainer_parameters": HuggingFaceTrainerParams(),
+        },
+        ValueError,
+    ),
+    (
+        "pvc creation failed",
+        {
+            "name": "tune_test",
+            "namespace": "pvc creation failed",
+            "model_provider_parameters": HuggingFaceModelParams(),
+            "dataset_provider_parameters": HuggingFaceDatasetParams(),
+            "trainer_parameters": HuggingFaceTrainerParams(),
+        },
+        RuntimeError,
+    ),
+    (
+        "valid flow with custom objective",
+        {
+            "name": "tune_test",
+            "namespace": "tune",
+            "objective": lambda x: x,
```

Comment:
Why this test case is valid if we test that parameters can be passed into objective function: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/utils/utils.py#L101 ?
E.g. we should be able to run this:
```py
objective({"param1": "value"})

Reply:
The objective function `"objective": lambda x: x` is an identity function, meaning it returns whatever input it receives unchanged. In the test case, This function will accept the input `{"param1": "value"}` and return it exactly as it was passed in.

Reply:
make sense, thanks.